src/core/Akka/Serialization/NewtonSoftJsonSerializer.cs

```diff
@@ -111,9 +111,10 @@ public NewtonSoftJsonSerializerSettings(bool encodeTypeNames, bool preserveObjec
     /// This is a special <see cref="Serializer"/> that serializes and deserializes javascript objects only.
     /// These objects need to be in the JavaScript Object Notation (JSON) format.
     /// </summary>
-    public class NewtonSoftJsonSerializer : Serializer
+    public partial class NewtonSoftJsonSerializer : Serializer
     {
-        private readonly JsonSerializer _serializer;
+        private readonly ThreadLocal<JsonSerializer> _serializer;
```

Comment:
I'm not a fan of ThreadLocal for a number of reasons, but I know that we have other challenges at present. @Aaronontheweb *is* there a polite way to get this fixed upstream?

Reply:
Getting this fixed upstream as in:

* Newtonsoft.Json itself?
* Akka.NET's overall serialization system?